CrashReport 3.1.3: StackOverflowError

**Describe the bug**
Recursive call to scan directory overflowed the stack

**Expected behavior**
Use File.walk() or `tailrec` recursive function

**Environment**
- Version of Kiwix Android : 5230103 
- Device : moto e5 play 
- OS version : 8.0 

**Logs**
```
Caused by: java.lang.StackOverflowError: 	at java.util.Arrays.copyOf(Arrays.java:3261)
	at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:125)
	at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:451)
	at java.lang.StringBuilder.append(StringBuilder.java:137)
	at java.io.UnixFileSystem.resolve(UnixFileSystem.java:105)
	at java.io.File.<init>(File.java:267)
	at java.io.File.listFiles(File.java:1212)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:1)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
	at org.kiwix.kiwixmobile.core.utils.files.FileSearch.scanDirectory(FileSearch.kt:5)
```

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileSearch.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileSearch.kt
@@ -71,17 +71,8 @@ class FileSearch @Inject constructor(private val context: Context) {
   private fun directoryRoots() =
     StorageDeviceUtils.getReadableStorage(context).map(StorageDevice::name)
 
-  private fun scanDirectory(directory: String): List<File> = File(directory).listFiles()
-    ?.fold(
-      mutableListOf(), { acc, file ->
-        acc.apply {
-          if (file.isDirectory) {
-            addAll(scanDirectory(file.path))
-          } else if (file.extension.isAny(*zimFileExtensions)) {
-            add(file)
-          }
-        }
-      }) ?: emptyList()
+  private fun scanDirectory(directory: String): List<File> =
+    File(directory).walk().filter { it.extension.isAny(*zimFileExtensions) }.toList()
 }
 
 internal fun String.isAny(vararg suffixes: String) =

